src/OpenCV/Features2d.hsc

```diff
@@ -285,6 +294,213 @@ orbDetectAndCompute orb img mbMask = unsafeWrapException $ do
 
           pure $ Right (V.fromList keypoints, relaxMat descriptors)
 
+--------------------------------------------------------------------------------
+-- BLOB - Binary Large OBject
+--------------------------------------------------------------------------------
+
+-- Internally, a SimpleBlobDetector is a pointer to a @cv::Ptr<cv::SimpleBlobDetector>@, which in turn points
+-- to an actual @cv::SimpleBlobDetector@ object.
+newtype SimpleBlobDetector = SimpleBlobDetector {unSimpleBlobDetector :: ForeignPtr C'Ptr_SimpleBlobDetector}
+
+type instance C SimpleBlobDetector = C'Ptr_SimpleBlobDetector
+
+instance WithPtr SimpleBlobDetector where
+    withPtr = withForeignPtr . unSimpleBlobDetector
+
+instance FromPtr SimpleBlobDetector where
+    fromPtr = objFromPtr SimpleBlobDetector $ \ptr ->
+                [CU.block| void {
+                  cv::Ptr<cv::SimpleBlobDetector> * simpleBlobDetector_ptr_ptr = $(Ptr_SimpleBlobDetector * ptr);
+                  simpleBlobDetector_ptr_ptr->release();
+                  delete simpleBlobDetector_ptr_ptr;
+                }|]
+
+data SimpleBlobDetectorParams
+   = SimpleBlobDetectorParams
+   { blob_blobColor :: !Word8
```

Comment:
Minor point: we align the curly braces `{` with the constructor name. So in this case we would it 2 places to the right (including the lines below).